dlt/destinations/impl/databricks/databricks.py

```diff
@@ -0,0 +1,336 @@
+            elif (
+                bucket_scheme == "s3"
+                and staging_credentials
+                and isinstance(staging_credentials, AwsCredentialsWithoutDefaults)
+            ):
+                s3_creds = staging_credentials.to_session_credentials()
+                credentials_clause = f"""WITH(CREDENTIAL(
+                AWS_ACCESS_KEY='{s3_creds["aws_access_key_id"]}',
+                AWS_SECRET_KEY='{s3_creds["aws_secret_access_key"]}',
+                AWS_SESSION_TOKEN='{s3_creds["aws_session_token"]}'
+                ))
+                """
+                from_clause = f"FROM '{bucket_path}'"
+            elif (
+                bucket_scheme in ["az", "abfs"]
+                and staging_credentials
+                and isinstance(staging_credentials, AzureCredentialsWithoutDefaults)
+            ):
+                # Explicit azure credentials are needed to load from bucket without a named stage
+                credentials_clause = f"""WITH(CREDENTIAL(AZURE_SAS_TOKEN='{staging_credentials.azure_storage_sas_token}'))"""
+                # Converts an az://<container_name>/<path> to abfss://<container_name>@<storage_account_name>.dfs.core.windows.net/<path>
+                # as required by snowflake
+                _path = bucket_url.path
+                bucket_path = urlunparse(
+                    bucket_url._replace(
+                        scheme="abfss",
+                        netloc=f"{bucket_url.netloc}@{staging_credentials.azure_storage_account_name}.dfs.core.windows.net",
+                        path=_path,
+                    )
+                )
+                from_clause = f"FROM '{bucket_path}'"
+            else:
+                raise LoadJobTerminalException(
+                    file_path,
+                    f"Databricks cannot load data from staging bucket {bucket_path}. Only s3 and azure buckets are supported",
+                )
+        else:
+            raise LoadJobTerminalException(
+                file_path,
+                "Cannot load from local file. Databricks does not support loading from local files. Configure staging with an s3 or azure storage bucket.",
+            )
+
+        # decide on source format, stage_file_path will either be a local file or a bucket path
+        source_format = "JSON"
+        if file_name.endswith("parquet"):
+            source_format = "PARQUET"
+
+        statement = f"""COPY INTO {qualified_table_name}
+            {from_clause}
+            {files_clause}
+            {credentials_clause}
+            FILEFORMAT = {source_format}
+            {format_options}
+            {copy_options}
+            """
+        client.execute_sql(statement)
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class DatabricksStagingCopyJob(SqlStagingCopyJob):
+    @classmethod
+    def generate_sql(
```

Comment:
i think some of the other destinations have very similar syntax for this, check bq and snowflake. maybe we can have a common stagingcopyjob shared between them, or a capability "supports_clone_command" so that we can just enable this on any sql destination and have a different syntax being emmitted from the SqlStagingCopyJob.

Reply:
syntax = sql_syntax

Reply:
Implemented it like this in https://github.com/dlt-hub/dlt/commit/64b7e2e5  . The `CLONE` syntax is exactly the same for bq/snowflake.  
https://github.com/dlt-hub/dlt/blob/64b7e2e583209146c0e13bbd532f22c66b0adea6/dlt/destinations/sql_jobs.py#L71-L127